### PR TITLE
Add data assimilation using usgs timeslices

### PR DESCRIPTION
**Pull request by jhrehanoaa**
https://github.com/NOAA-OWP/t-route/pull/258#discussion_r584912652 - From this discussion. 

RnR file can run with DA, but no matching columns in the parity check. I am wondering what the locations are for the timeslice files for this RnR? RnR had 'feature_id' instead of 'station_id' as well. Currently, this runs, but the parity test is comparing two totally different things essentially.

### CODE REVIEW

File: src/python_framework_v02/troute/nhd_io.py [hunk 1]
```diff
@@ -180,8 +181,8 @@ def get_ql_from_wrf_hydro_mf(qlat_files, index_col="feature_id", value_col="q_la
     2018-01-01 13:00:00 4186117     41.233807 -75.413895   0.006496
     ```
     """
-    filter_list = None
 
+    filter_list = None
```
Comment (jameshalgren) [inline src/python_framework_v02/troute/nhd_io.py:185]: I don't think we are using this parameter anywhere, so we can probably remove it.

**Comment by jameshalgren** [comment]
@jhrehanoaa 
Has this become a PR for integrating a test case for DA?

If so, could we rebase and start working on it from that angle?

File: src/python_framework_v02/troute/nhd_io.py [hunk 2]
```diff
@@ -322,27 +320,23 @@ def get_usgs_from_time_slices_csv(
         usgs_df.insert(i + 11, temp_name8, np.nan)
 
     usgs_df = usgs_df.interpolate(method="linear", axis=1)
-    usgs_df.drop(usgs_df[usgs_df.iloc[:,0] == -999999.000000].index , inplace=True)
+    usgs_df.drop(usgs_df[usgs_df.iloc[:, 0] == -999999.000000].index, inplace=True)
```
Comment (groutr) [inline src/python_framework_v02/troute/nhd_io.py:323]: If we are computing a mask, a call to drop is unnecessary
Assuming that first column of usgs_df is labeled "ColumnA"
```
usgs_df = usgs_df.loc[usgs_df["ColumnA"] == -999999]
```

File: src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py [hunk 3]
```diff
@@ -1044,18 +1044,19 @@ def main():
 
     if break_network_at_waterbodies:
         # Read waterbody parameters
-        waterbodies_df = nhd_io.read_waterbody_df(
-            waterbody_parameters, {"level_pool": wbodies.values()}
-        )
+        if wbodies != {}:
```
Comment (groutr) [inline src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py:1047]: `if wbodies:`

File: src/python_framework_v02/troute/nhd_io.py [hunk 4]
```diff
@@ -384,7 +378,7 @@ def get_usgs_from_time_slices_folder(
...
```
Comment (groutr) [inline src/python_framework_v02/troute/nhd_io.py:381]: See above comment. The call to .drop can be avoided here also.

**Comment by jameshalgren** [comment]
@jhrehanoaa -- I think this is a merge relic and am closing. Please reopen if this is still a relevant PR.

**Pull request closed without merging**